### DISCUSSION TITLE
hand and cell phone

### DISCUSSION THREAD

**Discussion by rickmaz** [vladmandic/human#487] [General]
I am using web version code here is is below. i am trying to get the right config and can not tell from the docs if it is even possible (not using node.js or backend) i would like to be able to detect if hands or fingers are in the webcam input and mobile phone if someone holding up image of a face on a phone for example: <script type="module">
        import Human from 'https://cdn.jsdelivr.net/npm/@vladmandic/human@3.2.1/dist/human.esm.js';

        document.addEventListener('DOMContentLoaded', async () => {
			
            const video = document.getElementById('video');
            const canvas = document.getElementById('canvas');
            const photo = document.getElementById('photo');
            const snap = document.getElementById('snap');
            const uploadForm = document.getElementById('uploadForm');
            const uploadButton = uploadForm.querySelector('button');
            const imageDataInput = document.getElementById('imageData');
            const instructions = document.getElementById('instructions');
			const ageStats = document.getElementById('ageStats');
            const progressBar = document.getElementById('progressBar');
            const progressCount = document.getElementById('progressCount');
			
			const DEBUG_MODE = false;
			toggleDebugInfo();
			
			const ANGLE_THRESHOLD_LOW = -0.25;
			const ANGLE_THRESHOLD_HIGH = 0.25;
			const GAZE_STRENGTH_THRESHOLD = 0.2;
			const MIN_ACCEPTABLE_DISTANCE = 0.65; // Adjust this value as needed
			const MAX_ACCEPTABLE_DISTANCE = 1.30; // Adjust this value as needed
			const MAX_DISTANCE_FOR_GOOD_PHOTO = 1.25;
						
			// Initialize session variables in JavaScript
            let ageLogs = [];
            let isSnapped = false;
            let step1_ageCount = 0;
            let step1_avgAge = 0;
            let step1_minAge = 999;
            let step1_maxAge = 0;
			
			let intervalId = null;
			let isPhotoTaken = false;
			
            const humanConfig = {
                modelBasePath: 'https://cdn.jsdelivr.net/npm/@vladmandic/human@3.2.1/models',
                async: true,
                 face: { 
				enabled: true,
				detector: { rotation: true },
				mesh: { enabled: true },
				iris: { enabled: true },
				description: { enabled: true },
				liveness: { enabled: true },
				antispoof: { enabled: true, threshold: 0.5 } },
                age: { enabled: true },
                filter: { enabled: true, flip: true }
            };
            
            const human = new Human(humanConfig);

**Comment by vladmandic** [comment]
if i understood correctly, then only partial fingers would be visible while holding the phone and palm of the hand would be obstructed by the phone - none of the hand detection models would detect hand with any kind of confidence in that case.
